src/UICommand.test.js

```diff
@@ -0,0 +1,87 @@
+import UICommand from './UICommand';
+import {Transaction} from 'prosemirror-state';
+import {createEditor, doc, p} from 'jest-prosemirror';
+
+describe('UICommand', () => {
+  const editor = createEditor(doc(p('<cursor>')));
+
+  beforeAll(() => {
+    // Create a spy on console (console.error in this case) and provide some mocked implementation
+    // In mocking global objects it's usually better than simple `jest.fn()`
+    // because you can `unmock` it in clean way doing `mockRestore`
+    jest.spyOn(console, 'error').mockImplementation(() => {});
+  });
+
+  afterAll(() => {
+    // Restore mock after all tests are done, so it won't affect other test suites
+    console.error.mockRestore();
+  });
+
+  it('isEnabled', () => {
+    const uiCmd = new UICommand();
+    const enabled = uiCmd.isEnabled(editor.state, editor.view);
+    expect(enabled).toEqual(false);
+  });
+
+  it('shouldRespondToUIEvent', () => {
+    const uiCmd = new UICommand();
+    const respond = uiCmd.shouldRespondToUIEvent({
+      type: UICommand.EventType.CLICK,
+    });
+    expect(respond).toEqual(true);
+  });
+
+  it('executeCustom', () => {
+    const uiCmd = new UICommand();
+    const tr = uiCmd.executeCustom(editor.state, editor.state.tr, 0, 0);
+    expect(tr.doc).toBe(editor.state.tr.doc);
+  });
+
+  it('isActive', () => {
+    const uiCmd = new UICommand();
+    const active = uiCmd.isActive(editor.state);
+    expect(active).toEqual(false);
+  });
+
+  it('renderLabel', () => {
+    const uiCmd = new UICommand();
+    const label = uiCmd.renderLabel(editor.state);
+    expect(label).toEqual(null);
+  });
+
+  it('isEnabled rejected', () => {
+    const uiCmd = new UICommand();
+    const mockWFUI = jest.fn();
+    mockWFUI.mockReturnValue(Promise.reject('this is error'));
+    uiCmd.waitForUserInput = mockWFUI;
+    const enabled = uiCmd.isEnabled(editor.state, editor.view);
+    expect(enabled).toEqual(false);
+  });
+
+  it('dryRunGet Transaction', () => {
+    const uiCmd = new UICommand();
+    const obj = uiCmd.dryRunEditorStateProxyGetter(editor.state, 'tr');
+    expect(obj).toBeInstanceOf(Transaction);
+
+    if (obj instanceof Transaction) {
+      const meta = obj.getMeta('dryrun');
+      expect(meta).toEqual(true);
+    }
+  });
+
+  it('dryRunGet Any', () => {
```

Comment:
Use test language to describe the branch being tested, and test (at minimum) every branch that will be encountered under normal circumstances.

```javascript
describe('dryRunEditorStateProxyGetter', () => {
    let tr: Transaction;
    let state;
    let uiCmd: UICommand;

    beforeEach(() => {
      tr = new Transaction({});
      tr.setMeta = jest.fn().mockImplementation(() => tr);
      // use same spy for both cases
      state = {tr, other: tr};
      uiCmd = new UICommand();
    });

    describe('when getting the transaction"', () => {
      it('should update transaction metadata', () => {
        const output = uiCmd.dryRunEditorStateProxyGetter(state, 'tr');

        expect(tr.setMeta).toHaveBeenCalled();
        expect(output).toBe(state.tr);
      });
    });

    describe('when getting other data', () => {
      it('should not update transaction metadata', () => {
        const output = uiCmd.dryRunEditorStateProxyGetter(state, 'other');

        expect(tr.setMeta).not.toHaveBeenCalled();
        expect(output).toBe(state.other);
      });
    });
  });
```

Reply:
Modified and Used this. Thanks!